index.test-d.ts

```diff
@@ -265,6 +265,12 @@ expectType<ExecaReturnValue>(await $`unicorns ${['foo', 'bar']}`);
 expectType<ExecaSyncReturnValue>($.sync`unicorns ${['foo', 'bar']}`);
 expectType<ExecaReturnValue>(await $`unicorns ${[1, 2]}`);
 expectType<ExecaSyncReturnValue>($.sync`unicorns ${[1, 2]}`);
+expectType<ExecaReturnValue>(await $`unicorns ${await $`echo foo`}`);
+expectError<ExecaReturnValue>(await $`unicorns ${$`echo foo`}`);
```

Comment:
This specific line might have an issue: ``$`echo foo` `` returns a promise, not an `ExecaReturnValue`, so that should not work?

Reply:
Have the test marked as `expectError`. Should I be handling this case in `parseExpression`? e.g.
```js
if (expression instanceof Promise) throw new TypeError('Invalid expression type')
```

Reply:
Oops, my bad, I did not notice it was using `expectError`!